### PR TITLE
CP-49451: Setup Github Action

### PR DESCRIPTION
**Pull request by initisan**
Two things have been done in this PR:
* Adding a CI process for terraform to build, lint and generate. 
>NOTE: UT is not integrated in CI as now the test leverage a non-mocked XS instance, so this part will do in the future
* Adding a release process to sign and release artifacts.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Terraform Provider CI workflow.
+name: Continuous Integration
+
+# This GitHub action runs your tests for each pull request and push.
+# Optionally, you can turn it on using a schedule for regular testing.
+on:
+  pull_request:
+    paths-ignore:
+      - "README.md"
```
Comment (xueqingz) [inline .github/workflows/ci.yml:9]: May be more here like "README.md", "LICENSE.md",...

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download and extract XenServer SDK
+        uses: ./.github/actions/setup-xenserver-sdk
+        with:
+          XENSERVER_SDK_URL: ${{ vars.XENSERVER_SDK_URL }}
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
+          go-version: ">=1.22.0"
```
Comment (xueqingz) [inline .github/workflows/ci.yml:33]: >=1.22.2

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      - run: go mod download
+      - run: go build -v .
+      - name: Run linters
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
+
+  generate:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (xueqingz) [inline .github/workflows/ci.yml:44]: I'm little confused here, why we do "Download and extract XenServer SDK" twice ? In build steps, we verified the build is success with SDK and check go lint. In generate  steps, we verify the document generated. why we need to separate these checks?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download and extract XenServer SDK
+        uses: ./.github/actions/setup-xenserver-sdk
+        with:
+          XENSERVER_SDK_URL: ${{ vars.XENSERVER_SDK_URL }}
```
Comment (xueqingz) [inline .github/workflows/ci.yml:29]: where this vars set? `vars.XENSERVER_SDK_URL `

Comment (initisan) [reply .github/workflows/ci.yml:29]: setup manually in the setting

**Pull request closed without merging**